testsuite/tests/kuadrant/reconciliation/cr_configuration/test_reconcile_limitador.py

```diff
@@ -0,0 +1,43 @@
+"""Tests for Kuadrant sub component - Limitador CR configuration via Kuadrant CR"""
+
+import pytest
+
+pytestmark = [pytest.mark.kuadrant_only]
+
+
+REPLICAS_1 = {"replicas": 1}
+REPLICAS_2 = {"replicas": 2}
+
+
+@pytest.fixture(scope="module")
+def kuadrant(request, kuadrant):
+    """Return the number of replicas for Limitador back to 1"""
+    request.addfinalizer(lambda: kuadrant.set_limitador_options(REPLICAS_1))
+    return kuadrant
+
+
+def test_spec_config(kuadrant):
+    """
+    Test:
+        - assert that Limitador deployment replicas is equal to 1
+        - change replicas field to 2 in Kuadrant CR
+        - assert that Kuadrant CR Limitador replicas is equal to 2
+        - assert that Limitador deployment replicas is equal to 2
+    """
+    assert kuadrant.model.spec.limitador.replicas == 1, ""
+    assert kuadrant.limitador_deployment.model.spec.replicas == 1, ""
+
+    kuadrant.set_limitador_options(REPLICAS_2)
+
+    assert kuadrant.model.spec.limitador["replicas"] == 2
```

Comment:
This is schizophrenic, you created API function that only passes down the dict for setting the options (set_limitador_options) but for getters you go directly to the model.

I would either remove the abstraction completely, or created a better one